crates/sui-framework/packages/dwallet-system/sources/eth_dwallet.move

```diff
@@ -39,8 +44,22 @@ module dwallet_system::eth_dwallet {
         dwallet_id: vector<u8>,
         data_slot: u64,
         proof: vector<u8>,
+        latest_ethereum_state: &LatestEthereumState,
+        current_wrapped_eth_state: &EthereumState,
     ): vector<MessageApproval> {
-        let is_valid = verify_message_proof(proof, message, dwallet_id, data_slot);
+        // Validate that the current Ethereum state from the user is the latest Ethereum state.
+        let current_ethereum_state_id = object::id(current_wrapped_eth_state);
+        let ethereum_state_id_from_latest = ethereum_state::get_ethereum_state_id_from_latest(latest_ethereum_state);
+        assert!(current_ethereum_state_id == ethereum_state_id_from_latest, 0x1);
```

Comment:
Maybe this check is too restrictive. It could be that someone else has updated `LatestEthereumState` whilst you are trying to approve a message and it will fail even though it should have passed. This would require you to retry. Depending on the frequency of Ethereum light client updates, it could be that this edge case would not be neglible.